framework/src/physics/PhysicsBase.C

```diff
@@ -477,3 +495,241 @@ PhysicsBase::addPetscPairsToPetscOptions(
         *this,
         po);
 }
+
+bool
+PhysicsBase::isVariableFV(const VariableName & var_name) const
+{
+  const auto var = &_problem->getVariable(0, var_name);
+  if (const auto & field_var = dynamic_cast<MooseVariableFieldBase *>(var))
+    return field_var->isFV();
+  return false;
+}
+
+bool
+PhysicsBase::isVariableScalar(const VariableName & var_name) const
+{
+  return _problem->hasScalarVariable(var_name);
+}
+
+bool
+PhysicsBase::shouldCreateVariable(const VariableName & var_name,
+                                  const std::vector<SubdomainName> & blocks,
+                                  const bool error_if_aux)
+{
+  if (!variableExists(var_name, error_if_aux))
+    return true;
+  // check block restriction
+  auto & var = _problem->getVariable(0, var_name);
+  const bool not_block_restricted =
+      (std::find(blocks.begin(), blocks.end(), "ANY_BLOCK_ID") != blocks.end()) ||
+      allMeshBlocks(blocks);
+  if (!var.blockRestricted() || (!not_block_restricted && var.hasBlocks(blocks)))
+    return false;
+
+  // This is an edge case, which might warrant a warning
+  if (allMeshBlocks(var.blocks()) && not_block_restricted)
+    return false;
+  else
+    mooseError("Variable '" + var_name + "' already exists with subdomain restriction '" +
+               Moose::stringify(var.blocks()) + "' which does not include the subdomains '" +
+               Moose::stringify(blocks) + "', required for this Physics.");
+}
+
+bool
+PhysicsBase::shouldCreateIC(const VariableName & var_name,
+                            const std::vector<SubdomainName> & blocks,
+                            const bool ic_is_default_ic,
+                            const bool error_if_already_defined) const
+{
+  // Handle recover
+  if (ic_is_default_ic && (_app.isRestarting() || _app.isRecovering()))
+    return false;
+  // do not set initial conditions if we are loading fields from the mesh file
+  if (getParam<bool>("initialize_variables_from_mesh_file"))
+    return false;
+  // Different type of ICs, not block restrictable
+  mooseAssert(!isVariableScalar(var_name), "shouldCreateIC not implemented for scalar variables");
+
+  // Deal with block restriction
+  bool has_all_blocks;
+  std::set<SubdomainName> blocks_covered;
```

Comment:
Can you use IDs instead of names? This would make it unnecessary to do the set of IDs to vec of IDs to vec of names to set of names conversion. And also its easier/faster to compare integers than strings

Reply:
ok let me try that!